plumbing/cst/chain_state.go

```diff
@@ -28,8 +27,8 @@ type chainReader interface {
 // ChainStateProvider composes a chain and a state store to provide access to
 // the state (including actors) derived from a chain.
 type ChainStateProvider struct {
-	reader chainReader         // Provides chain blocks and tipsets.
-	cst    *hamt.CborIpldStore // Provides state trees.
+	reader chainReader         // Provides chain tipsets and state roots.
+	cst    *hamt.CborIpldStore // Provides chain blocks and state trees.
```

Comment:
It's unfortunate that the ChainStateProvider is required to provide blocks at all. It's worth a quick investigation to see if you can fix the callers to not need that, then revert this documentation not to reference chain blocks at all.

Reply:
`go-filecoin show block` needs this right now so I'm leaving it for this PR.